spec/dataframe_spec.rb

```diff
@@ -789,9 +789,67 @@
     context Daru::MultiIndex do
       pending
     end
+
+    context Daru::CategoricalIndex do
+      let(:idx) { Daru::CategoricalIndex.new [:a, 1, :a, 1, :c] }
+      let(:dv) { Daru::Vector.new ['x', 'y'] }
+      let(:df) do
+        Daru::DataFrame.new({
+          a: 'a'..'e',
+          b: 1..5
+        }, index: idx)
+      end
+
+      context "modify exiting row" do
+        context "single category" do
+          subject { df }
+          before { df.row[:a] = dv }
+
+          it { is_expected.to be_a Daru::DataFrame }
+          its(:index) { is_expected.to eq idx }
+          its(:vectors) { is_expected.to eq Daru::Index.new [:a, :b] }
+          its(:a) { Daru::Vector.new ['x', 'b', 'x', 'x', 'e'] }
+          its(:b) { Daru::Vector.new ['y', 2, 'y', 'y', 5] }
```

Comment:
I think it should be `['y', 2, 'y', 4, 5]` and `['x', 'b', 'x', 'd', 'e']`, i.e., one too many 'x's and 'y's.